scalding-core/src/main/scala/com/twitter/scalding/typed/TypedPipe.scala

```diff
@@ -143,17 +143,16 @@ object TypedPipe extends Serializable {
       }
   }
 
-  private val identityOrdering: OrderedSerialization[Int] = {
+  private case object IdentityOrdering extends OrderedSerialization[Int] with EquivSerialization[Int] {
     val delegate = BinaryOrdering.ordSer[Int]
-    new OrderedSerialization[Int] {
-      override def compareBinary(a: InputStream, b: InputStream): Result = delegate.compareBinary(a, b)
-      override def compare(x: Int, y: Int): Int = delegate.compare(x, y)
-      override def dynamicSize(t: Int): Option[Int] = delegate.dynamicSize(t)
-      override def write(out: OutputStream, t: Int): Try[Unit] = delegate.write(out, t)
-      override def read(in: InputStream): Try[Int] = delegate.read(in)
-      override def staticSize: Option[Int] = delegate.staticSize
-      override def hash(x: Int): Int = x
-    }
+    
+    override def compareBinary(a: InputStream, b: InputStream): Result = delegate.compareBinary(a, b)
+    override def compare(x: Int, y: Int): Int = delegate.compare(x, y)
+    override def dynamicSize(t: Int): Option[Int] = delegate.dynamicSize(t)
+    override def write(out: OutputStream, t: Int): Try[Unit] = delegate.write(out, t)
+    override def read(in: InputStream): Try[Int] = delegate.read(in)
+    override def staticSize: Option[Int] = delegate.staticSize
+    override def hash(x: Int): Int = x
```

Comment:
small note: why don't we do `delegate.hash(x)` here? seems weird that we delegate through the others. Maybe a good reason, we have had weird bugs related to non-uniform partitioning on this method before, and this may be related to fixing that, but it would be nice if we had commented that.